internal/daemon/controller/handlers/accounts/account_service.go

```diff
@@ -132,70 +138,307 @@ func NewService(ctx context.Context, pwRepo common.PasswordAuthRepoFactory, oidc
 	case ldapRepo == nil:
 		return Service{}, errors.New(ctx, errors.InvalidParameter, op, "missing ldap repository")
 	}
-	return Service{pwRepoFn: pwRepo, oidcRepoFn: oidcRepo, ldapRepoFn: ldapRepo}, nil
+	if maxPageSize == 0 {
+		maxPageSize = uint(defaultMaxPageSize)
+	}
+	return Service{
+		pwRepoFn:    pwRepo,
+		oidcRepoFn:  oidcRepo,
+		ldapRepoFn:  ldapRepo,
+		maxPageSize: maxPageSize,
+	}, nil
 }
 
 // ListAccounts implements the interface pbs.AccountServiceServer.
 func (s Service) ListAccounts(ctx context.Context, req *pbs.ListAccountsRequest) (*pbs.ListAccountsResponse, error) {
+	const op = "accounts.(Service).ListAccounts"
 	if err := validateListRequest(ctx, req); err != nil {
 		return nil, err
 	}
-	_, authResults := s.parentAndAuthResult(ctx, req.GetAuthMethodId(), action.List)
+	authMethodId := req.GetAuthMethodId()
+	_, authResults := s.parentAndAuthResult(ctx, authMethodId, action.List)
 	if authResults.Error != nil {
 		return nil, authResults.Error
 	}
-	ul, err := s.listFromRepo(ctx, req.GetAuthMethodId())
-	if err != nil {
-		return nil, err
-	}
-	if len(ul) == 0 {
-		return &pbs.ListAccountsResponse{}, nil
-	}
 
 	filter, err := handlers.NewFilter(ctx, req.GetFilter())
 	if err != nil {
 		return nil, err
 	}
-	finalItems := make([]*pb.Account, 0, len(ul))
 
-	res := perms.Resource{
-		ScopeId: authResults.Scope.Id,
-		Type:    resource.Account,
-		Pin:     req.GetAuthMethodId(),
-	}
-	for _, acct := range ul {
-		res.Id = acct.GetPublicId()
-		authorizedActions := authResults.FetchActionSetForId(ctx, acct.GetPublicId(), IdActions[subtypes.SubtypeFromId(domain, acct.GetPublicId())], requestauth.WithResource(&res)).Strings()
-		if len(authorizedActions) == 0 {
-			continue
+	switch subtypes.SubtypeFromId(domain, authMethodId) {
+	case ldap.Subtype:
+		ldapRepo, err := s.ldapRepoFn()
+		if err != nil {
+			return nil, errors.Wrap(ctx, err, op)
 		}
+		listAccountsFn := func(prevPageLast *ldap.Account, refreshToken *pbs.ListRefreshToken, limit int) ([]*ldap.Account, error) {
```

Comment:
We might be able to use `auth.Account` as the type, instead of `*ldap.Account`, which would allow us to reuse some of these functions (the filterAndConvert, I think).